### PR TITLE
[Fix] fix the onnx exportation for yoloxpose in mmpose

### PR DESCRIPTION
**Pull request by Ben-Louis**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

In the latest version of mmpose, YOLOX-Pose models have been moved from projects to the core package. As a result, the model structure has changed and the old implementation in mmdeploy is no longer compatible.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by irexyc — APPROVED** [review]

File: mmdeploy/codebase/mmpose/deploy/pose_detection_model.py [hunk 1]
```diff
@@ -145,11 +147,11 @@ def pack_result(self,
             # convert keypoint coordinates from input space to image space
             if convert_coordinate:
                 input_size = data_sample.metainfo['input_size']
-                bbox_centers = gt_instances.bbox_centers
-                bbox_scales = gt_instances.bbox_scales
+                input_center = data_sample.metainfo['input_center']
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmpose/deploy/pose_detection_model.py:150]: other models like `hrnet` fails in here:
```
Traceback (most recent call last):
  File "/root/workspace/mmdeploy/mmdeploy/utils/utils.py", line 41, in target_wrapper
    result = target(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/apis/visualize.py", line 72, in visualize_model
    result = model.test_step(model_inputs)[0]
  File "/usr/local/lib/python3.8/dist-packages/mmengine/model/base_model/base_model.py", line 145, in test_step
    return self._run_forward(data, mode='predict')  # type: ignore
  File "/usr/local/lib/python3.8/dist-packages/mmengine/model/base_model/base_model.py", line 340, in _run_forward
    results = self(**data, mode=mode)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/codebase/mmpose/deploy/pose_detection_model.py", line 116, in forward
    results = self.pack_result(preds, data_samples)
  File "/root/workspace/mmdeploy/mmdeploy/codebase/mmpose/deploy/pose_detection_model.py", line 150, in pack_result
    input_center = data_sample.metainfo['input_center']
KeyError: 'input_center'

```

Comment (Ben-Louis) [reply mmdeploy/codebase/mmpose/deploy/pose_detection_model.py:150]: The post-processing modification for top-down models is now compatible with the `dev-1.x` branch of mmpose. This branch will be merged into the `main` branch soon

File: mmdeploy/codebase/mmpose/models/heads/yolox_pose_head.py [hunk 2]
```diff
@@ -11,18 +10,18 @@
 from mmdeploy.utils import Backend, get_backend
 
 
-@FUNCTION_REWRITER.register_rewriter(func_name='models.yolox_pose_head.'
-                                     'YOLOXPoseHead.predict')
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmpose.models.heads.hybrid_heads.'
+    'yoloxpose_head.YOLOXPoseHead.forward')
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmpose/models/heads/yolox_pose_head.py:15]: Could you add regression config to test it?
1. add in `mmyolo.yml`, could refer to https://github.com/open-mmlab/mmdeploy/blob/4cf0f9291d846d660744867c192e51f46a63a761/tests/regression/mmpose.yml#L130
2. test it with `python3 tools/regression_test.py --codebase mmpose --models YOLOX-Pose  --backends onnxruntime`

Comment (Ben-Louis) [reply mmdeploy/codebase/mmpose/models/heads/yolox_pose_head.py:15]: I ran the regression test and here is the generated report 
截屏2023-09-28 16 40 50
Does it look right?

Comment (RunningLeon) [reply mmdeploy/codebase/mmpose/models/heads/yolox_pose_head.py:15]: Looks good

File: mmdeploy/codebase/mmpose/deploy/pose_detection_model.py [hunk 3]
```diff
@@ -98,10 +98,12 @@ def forward(self,
         inputs = inputs.contiguous().to(self.device)
         batch_outputs = self.wrapper({self.input_name: inputs})
         batch_outputs = self.wrapper.output_to_list(batch_outputs)
-        if self.model_cfg.model.type == 'YOLODetector':
-            return self.pack_yolox_pose_result(batch_outputs, data_samples)
 
         codec = self.model_cfg.codec
+
+        if codec.type == 'YOLOXPoseAnnotationProcessor':
```
Comment (RunningLeon) [inline mmdeploy/codebase/mmpose/deploy/pose_detection_model.py:104]: @Ben-Louis hi, this two lines should be put after line 108 as the codec maybe list of dict as in [mspn](https://github.com/open-mmlab/mmpose/blob/8102383761e877317825ef0d5e234ca4d584e5bc/configs/body_2d_keypoint/topdown_heatmap/coco/td-hm_mspn50_8xb32-210e_coco-256x192.py#L35)
```
Traceback (most recent call last):
  File "/root/workspace/mmdeploy/mmdeploy/utils/utils.py", line 41, in target_wrapper
    result = target(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/apis/visualize.py", line 72, in visualize_model
    result = model.test_step(model_inputs)[0]
  File "/usr/local/lib/python3.8/dist-packages/mmengine/model/base_model/base_model.py", line 145, in test_step
    return self._run_forward(data, mode='predict')  # type: ignore
  File "/usr/local/lib/python3.8/dist-packages/mmengine/model/base_model/base_model.py", line 340, in _run_forward
    results = self(**data, mode=mode)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/codebase/mmpose/deploy/pose_detection_model.py", line 104, in forward
    if codec.type == 'YOLOXPoseAnnotationProcessor':
AttributeError: 'list' object has no attribute 'type'
```

Comment (Ben-Louis) [reply mmdeploy/codebase/mmpose/deploy/pose_detection_model.py:106]: Thanks. This bug has been fixed in the latest commit

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**